code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -213,6 +213,16 @@
 		PDA.f_lum = 0
 		PDA.update_icon()
 		visible_message("<span class='danger'>The light in [PDA] shorts out!</span>")
+	else if(istype(O, /obj/item/gun))
+		var/obj/item/gun/weapon = O
+		if(weapon.gun_light)
+			var/obj/item/flashlight/seclite/light = weapon.gun_light
+			light.forceMove(get_turf(weapon))
+			light.burn()
+			weapon.gun_light = null
+			weapon.update_gunlight()
+			QDEL_NULL(weapon.alight)
+			visible_message("<span class='danger'>[light] on [O] flickers out and disintegrates!</span>")
```

Comment:
You need to run this visible message before `light` is burned and qdeleted.

Reply:
![imagen](https://user-images.githubusercontent.com/3855300/74923318-d9c39d00-53d0-11ea-9b8b-ada3ed15dd45.png)
works fine on my end

Reply:
the qdelete is for the action button